kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h

```diff
@@ -0,0 +1,434 @@
+/// Dirichlet boundary condition for rUnknownVar at velocity inlets/outles is essential to be set for this solver since it is based on the flux
+template< unsigned int TDim, unsigned int TNumNodes>
+class LevelSetConvectionElementSimplexAlgebraicStabilization
+    : public Element
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(LevelSetConvectionElementSimplexAlgebraicStabilization);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    LevelSetConvectionElementSimplexAlgebraicStabilization() : Element()
+    {}
+
+    LevelSetConvectionElementSimplexAlgebraicStabilization(IndexType NewId, GeometryType::Pointer pGeometry)
+    : Element(NewId, pGeometry)
+    {}
+
+    LevelSetConvectionElementSimplexAlgebraicStabilization(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
+    : Element(NewId, pGeometry, pProperties)
+    {}
+
+    /// Destructor.
+    ~LevelSetConvectionElementSimplexAlgebraicStabilization() override {};
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override
+    {
+        KRATOS_TRY
+        return Element::Pointer(new LevelSetConvectionElementSimplexAlgebraicStabilization(NewId, GetGeometry().Create(ThisNodes), pProperties));
+        KRATOS_CATCH("");
+    }
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        if (rLeftHandSideMatrix.size1() != TNumNodes)
+            rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false); //false says not to preserve existing storage!!
+
+        if (rRightHandSideVector.size() != TNumNodes)
+            rRightHandSideVector.resize(TNumNodes, false); //false says not to preserve existing storage!!
+
+        const double delta_t = rCurrentProcessInfo[DELTA_TIME];
+        const double dt_inv = 1.0 / delta_t;
+        const double theta = rCurrentProcessInfo.Has(TIME_INTEGRATION_THETA) ? rCurrentProcessInfo[TIME_INTEGRATION_THETA] : 0.5;
+
+        ConvectionDiffusionSettings::Pointer my_settings = rCurrentProcessInfo.GetValue(CONVECTION_DIFFUSION_SETTINGS);
+        const Variable<double>& rUnknownVar = my_settings->GetUnknownVariable();
+        const Variable<array_1d<double, 3 > >& rConvVar = my_settings->GetConvectionVariable();
+        const Variable<array_1d<double, 3 > >& rGradVar = my_settings->GetGradientVariable();
+
+        //getting data for the given geometry
+        BoundedMatrix<double, TNumNodes, TDim > DN_DX;
+        array_1d<double, TNumNodes > N;
+        double Volume;
+        GeometryUtils::CalculateGeometryData(GetGeometry(), DN_DX, N, Volume);
```

Comment:
(comment for the future) 

We are using in there the `GeometryUtils` to obtain only the gradients. This happens quite often. I think we should add another method to return only these.